saleor/discount/tasks.py

```diff
@@ -156,3 +169,21 @@ def fetch_promotion_variants_and_product_ids(promotions: "QuerySet[Promotion]"):
         Exists(variants.filter(product_id=OuterRef("id")))
     )
     return promotion_id_to_variants, list(products.values_list("id", flat=True))
+
+
+@app.task
+def clear_promotion_rule_variants_task():
+    """Clear all promotion rule variants."""
+    promotions = Promotion.objects.expired()
+    rules = PromotionRule.objects.filter(
+        Exists(promotions.filter(id=OuterRef("promotion_id")))
+    )
```

Comment:
I see that `expired()` method do `self.filter(end_date__lt=date, start_date__lt=date)` , these fields are not indexed so that will cause a possibly large sequence scan. I don't know much about promotions, if it's unlikely that users will have a lot of promotion objects, this is perfectly fine, otherwise, I would advise measuring it on a larger dataset to see how it performs.

Reply:
I added indexes to `start_date` and `end_date`